server/src/main/java/org/elasticsearch/http/AbstractHttpServerTransport.java

```diff
@@ -388,9 +388,7 @@ protected void serverAcceptedChannel(HttpChannel httpChannel) {
         try {
             rlock.lock();
             if (shuttingDown) {
-                logger.warn("server accepted channel after shutting down");
-                httpChannel.close();
-                return;
+                throw new IllegalStateException("server accepted channel after shutting down");
```

Comment:
The error message is a bit cryptic, could we improve it? Something like `Cannot accept new server channel while shutting down`

Reply:
Good call, done.